storage/badger/operation/approvals.go

```diff
@@ -1,31 +1,33 @@
 package operation
 
 import (
-	"github.com/dgraph-io/badger/v2"
-
 	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/storage"
 )
 
 // InsertResultApproval inserts a ResultApproval by ID.
-func InsertResultApproval(approval *flow.ResultApproval) func(*badger.Txn) error {
-	return insert(makePrefix(codeResultApproval, approval.ID()), approval)
+// The same key (`approval.ID()`) necessitates that the value (full `approval`) is
+// also identical (otherwise, we would have a successful pre-image attack on our
+// cryptographic hash function). Therefore, concurrent calls to this function are safe.
+func InsertResultApproval(approval *flow.ResultApproval) func(storage.Writer) error {
+	return insertW(makePrefix(codeResultApproval, approval.ID()), approval)
 }
 
 // RetrieveResultApproval retrieves an approval by ID.
-func RetrieveResultApproval(approvalID flow.Identifier, approval *flow.ResultApproval) func(*badger.Txn) error {
-	return retrieve(makePrefix(codeResultApproval, approvalID), approval)
+func RetrieveResultApproval(approvalID flow.Identifier, approval *flow.ResultApproval) func(storage.Reader) error {
+	return retrieveR(makePrefix(codeResultApproval, approvalID), approval)
 }
 
 // IndexResultApproval inserts a ResultApproval ID keyed by ExecutionResult ID
 // and chunk index. If a value for this key exists, a storage.ErrAlreadyExists
 // error is returned. This operation is only used by the ResultApprovals store,
 // which is only used within a Verification node, where it is assumed that there
 // is only one approval per chunk.
```

Comment:
```suggestion
// is only one approval per chunk.
// CAUTION: Use of this function must be synchronized by storage.ResultApprovals.
```
I'd like to communicate which low-level functions must be synchronized by a higher-level storage procedure. Adding documentation and using `Unsafe...` naming seems like a relatively easy way to do that.